### PR TITLE
Add URI singleton methods

### PR DESCRIPTION
**Pull request by ybiquitous**
This change adds signatures of the `URI` module singleton methods (and constants).
This also adds `URI:RFC*` modules and classes to pass the test but their content are empty still.

I refer to the following resources:
- https://ruby-doc.org/stdlib-2.7.2/libdoc/uri/rdoc/URI.html
- https://github.com/ruby/ruby/blob/v2_7_2/lib/uri/common.rb
- https://github.com/ruby/ruby/blob/v2_7_2/test/uri/test_common.rb

### CODE REVIEW

**Comment by ybiquitous** [comment]
Oh, I forget to add the `Kernel::URI` method. I'll add it soon.

**Comment by ybiquitous** [comment]
Done. (squashed)

File: stdlib/uri/0/common.rbs [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+  #     ary = URI.decode_www_form("a=1&a=2&b=3")
+  #     ary                   #=> [['a', '1'], ['a', '2'], ['b', '3']]
+  #     ary.assoc('a').last   #=> '1'
+  #     ary.assoc('b').last   #=> '3'
+  #     ary.rassoc('a').last  #=> '2'
+  #     Hash[ary]             #=> {"a"=>"2", "b"=>"3"}
+  #
+  # See URI.decode_www_form_component, URI.encode_www_form.
+  #
+  def self.decode_www_form: (String str, ?encoding enc, ?isindex: bool, ?use__charset_: bool, ?separator: String) -> Array[[ String, String ]]
```
Comment (soutaro) [inline stdlib/uri/0/common.rbs:119]: It seems like `isindex` and `use__charset_` accepts any value, not only `true` and `false`.
`boolish` would be better?

**Review by soutaro — CHANGES_REQUESTED** [review]
`boolish` looks better to me.

File: stdlib/uri/0/common.rbs (hunk 1 above)
Comment (ybiquitous) [reply stdlib/uri/0/common.rbs:119]: Surely, `boolish` is more flexible. I'll fix it soon.

Comment (ybiquitous) [reply stdlib/uri/0/common.rbs]: Fixed via 708bbf4

**Comment by ybiquitous** [comment]
By the way, does adding a Goodcheck rule below make sense? 👇 

```yaml
  - id: prefer_boolish_over_bool
    pattern:
      - regexp: '\([^(]*\bbool\b[^\n]*\)'
      - token: "-> bool }"
    message: |
      Prefer `boolish` over `bool` for method arguments and block return values

      See the doc below:
      https://github.com/ruby/rbs/blob/78d04a2db0f1c4925d2b13c2939868edf9465d6c/docs/syntax.md#bool-or-boolish
    glob:
      - "**/*.rbs"
    justification:
      - When the method arguments or block return values require `true` or `false` strictly.
    pass:
      - "(arg: boolish)"
      - "{ () -> boolish }"
    fail:
      - "(arg: bool)"
      - "{ () -> bool }"
```

According to <https://github.com/ruby/rbs/blob/78d04a2db0f1c4925d2b13c2939868edf9465d6c/docs/syntax.md#bool-or-boolish>.

E.g.

```console
$ be goodcheck check -R prefer_boolish_over_bool core
core/complex.rbs:168:14: Prefer `boolish` over `bool` for method arguments and block return values
  def clone: (?freeze: bool) -> self
             ^~~~~~~~~~~~~~~
core/exception.rbs:193:21: Prefer `boolish` over `bool` for method arguments and block return values
  def full_message: (?highlight: bool, ?order: :top | :bottom) -> String
                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
core/float.rbs:174:14: Prefer `boolish` over `bool` for method arguments and block return values
  def clone: (?freeze: bool) -> self
             ^~~~~~~~~~~~~~~
```

File: stdlib/uri/0/common.rbs (hunk 1 above)
Comment (ybiquitous) [reply stdlib/uri/0/common.rbs]: I've rebased `master` and squashed to one commit.

**Comment by ybiquitous** [comment]
I've open PR #531 to add a Goodcheck rule as a test, but please close the PR if you don't want.

**Review by soutaro — APPROVED** [review]
Thanks!
